packages/http-client-java/generator/http-client-generator-mgmt/src/main/java/com/microsoft/typespec/http/client/generator/mgmt/template/FluentMethodMockTestTemplate.java

```diff
@@ -72,6 +72,7 @@ public void write(ClientMethodInfo info, JavaFile javaFile) {
             fluentReturnType = FluentUtils.getValueTypeFromResponseType(fluentReturnType);
         }
         final boolean hasReturnValue = fluentReturnType.asNullable() != ClassType.VOID;
+        final boolean isCheckExistence = fluentReturnType.asNullable() == ClassType.BOOLEAN;
```

Comment:
Maybe add a comment about this, or see if we can also condition on HTTP method type

Return type is boolean not necessary means it is 204/404 HEAD. Though in mgmt there likely won't have any API return JSON boolean.

Reply:
updated